chaco/candle_plot.py

```diff
@@ -1,191 +1,19 @@
-                return None
-            else:
-                return index
-        else:
-            screen = self.index_mapper.map_screen(index_data[0])
-            if threshold > 0 and abs(screen - screen_pt[0]) >= threshold:
-                return None
-            else:
-                return index
-
-    def _gather_points(self):
-        index = self.index.get_data()
-        mask = broaden(self.index_range.mask_data(index))
-
-        if not mask.any():
-            self._cached_data_pts = []
-            self._cache_valid = True
-            return
-
-        data_pts = [compress(mask, index)]
-
-        for v in (
-            self.min_values,
-            self.bar_min,
-            self.center_values,
-            self.bar_max,
-            self.max_values,
-        ):
-            if v is None or len(v.get_data()) == 0:
-                data_pts.append(None)
-            else:
-                data_pts.append(compress(mask, v.get_data()))
-
-        self._cached_data_pts = data_pts
-        self._cache_valid = True
-
-    def _draw_plot(self, gc, view_bounds=None, mode="normal"):
-        self._gather_points()
-        if len(self._cached_data_pts) == 0:
-            return
-
-        index = self.index_mapper.map_screen(self._cached_data_pts[0])
-        if len(index) == 0:
-            return
-
-        vals = []
-        for v in self._cached_data_pts[1:]:
-            if v is None:
-                vals.append(None)
-            else:
-                vals.append(self.value_mapper.map_screen(v))
-
-        # Compute lefts and rights from self.index, which represents bin
-        # centers.
-        if len(index) == 1:
-            width = 5.0
-        else:
-            width = (index[1:] - index[:-1]).min() / 2.5
-        left = index - width
-        right = index + width
-
-        with gc:
-            gc.clip_to_rect(self.x, self.y, self.width, self.height)
-            self._render(gc, left, right, *vals)
-
-    def _get_value(self):
-        if self.center_values is not None:
-            return self.center_values
-        elif self.bar_min is not None:
-            return self.bar_min
-        elif self.bar_max is not None:
-            return self.bar_max
+# (C) Copyright 2006-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import warnings
+
+from chaco.plots.candle_plot import CandlePlot  # noqa: F401
+
+warnings.warn(
+    "This module has been moved to sit in chaco/plots and this stub module has"
+    " been kept for backwards compatibility. Importing from this module is"
+    " deprecated, please import needed objects from chaco.api instead",
+    DeprecationWarning
+)
```

Comment:
Note: "Import-time deprecation warnings risk being hidden from test runs."
We may want to update how we are raising these deprecation warnings...

Reply:
We should also update the warning message to be explicit that these modules will be removed in chaco V6 or V7 once that is decided on